### PR TITLE
javadsl should use java types #23251

### LINKED ISSUE #23251: javadsl API oversight when using primitive types in collections or functions
e.g.
```
def create[T](outputCount: Int, partitioner: function.Function[T, Int]): Graph[UniformFanOutShape[T, T], NotUsed]
```

becomes `Function<T, Object>`

must use `Integer`

Might be similar problems in many places, strange that none has been reported.

### PR DESCRIPTION
**Pull request by patrox**
Fixes #23251

For now all methods within `akka.stream` javadsl have been verified
and updated where applicable.

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

**Review by ktoso — APPROVED** [review]
LGTM, thanks a lot 👍

**Comment by ktoso** [comment]
Such changes should be good to merge, as previously the API would have been not properly usable, and the changes are in a type param, so it should be binary fine as well

**Comment by johanandren** [comment]
Seems like a test broke, will need an update as well:

```
akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java:638: method batchWeighted in class akka.stream.javadsl.Flow<In,Out,Mat> cannot be applied to given types;
[error]   required: long,akka.japi.function.Function<java.lang.String,java.lang.Long>,akka.japi.function.Function<java.lang.String,S>,akka.japi.function.Function2<S,java.lang.String,S>
[error]   found: long,<anonymous akka.japi.function.Function<java.lang.String,java.lang.Object>>,<anonymous akka.japi.function.Function<java.lang.String,java.lang.String>>,<anonymous akka.japi.function.Function2<java.lang.String,java.lang.String,java.lang.String>>
[error]   reason: cannot infer type-variable(s) S
...
```

**Comment by patrox** [comment]
@johanandren I'll update the test as well.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

File: akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java [hunk 1]
```diff
@@ -635,9 +635,9 @@ public String apply(String aggr, String in) throws Exception {
   public void mustBeAbleToUseBatchWeighted() throws Exception {
     final TestKit probe = new TestKit(system);
     final List<String> input = Arrays.asList("A", "B", "C");
-    final Flow<String, String, NotUsed> flow = Flow.of(String.class).batchWeighted(3L, new Function<String, Object>() {
+    final Flow<String, String, NotUsed> flow = Flow.of(String.class).batchWeighted(3L, new Function<String, java.lang.Long>() {
       @Override
-      public Object apply(String s) throws Exception {
```
Comment (patriknw) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java]: wow, so we even had test that accepted this oversight

Comment (ktoso) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java]: whoa sux :<

Comment (patrox) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java]: Yup, seems like there was some misunderstanding in the way how it should work.
But, now the type-safety-net has been restored :)

**Pull request merged**